benchmark-service/benchmark/utils.py

```diff
@@ -0,0 +1,36 @@
+from google.cloud import storage
+import re
+
+
+def get_geometric_mean(prod_of_means, num_of_means):
+    return prod_of_means ** (1.0 / num_of_means)
+
+
+class ReadGoogleStorage:
+    storage_client = storage.Client()
+
+    FILE_PATH_REGEX = re.compile(r'gs://((?P<bucket>[^/]+)/)((?P<user>[^/]+)/)((?P<version>[^-]+)-)((?P<sha>[^-]+))(-(?P<tag>[^\.]+))?\.json')
+
+    @staticmethod
+    def remove_prefix(text, prefix):
+        if text.startswith(prefix):
+            return text[len(prefix):]
+        return text
+
+    def parse_file_path(self, name):
+        match = self.FILE_PATH_REGEX.fullmatch(name)
```

Comment:
I think this function can return a tuple of (bucket, path) by parsing the groupdict. Then you don't need the remove_prefix function.